crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1818,6 +1929,21 @@ impl<'db> Type<'db> {
                     CallOutcome::callable(len_ty.unwrap_or(normal_return_ty))
                 }
 
+                Some(KnownFunction::AssertType) => {
+                    let normal_return_ty = function_type.signature(db).return_ty;
+
+                    let [actual_type, asserted] = arg_types else {
+                        // TODO: Emit a diagnostic
+                        return CallOutcome::callable(normal_return_ty);
+                    };
+                    let Ok(asserted_type) = asserted.in_type_expression(db) else {
+                        // TODO: Emit a diagnostic
+                        return CallOutcome::callable(normal_return_ty);
+                    };
+
+                    CallOutcome::asserted(normal_return_ty, *actual_type, asserted_type)
+                }
```

Comment:
These todos can be solved by rebasing this on top of call-checking, which has landed, and https://github.com/astral-sh/ruff/pull/15103, which hopefully will land soon, and has many examples of similar methods.

The one thing we'll need to do that https://github.com/astral-sh/ruff/pull/15103 doesn't yet do is provide a way to say that some arguments of a function must be interpreted as type expressions, and others as value expressions.